### PR TITLE
Added reddit example (#29)

### PR DESCRIPTION
**Pull request by biplobsd**
Closed: #29 

## Summary
Submit a post on subreddit.

## Payload
```json
{
  "sr": "r/bottest",
  "text": "🔥",
  "title": "Hello"
}
```

## Screen record
[reddit-example.webm](https://github.com/triggerdotdev/api-reference/assets/43641536/aff677e1-abfb-492d-acac-e946e6f10e59)

### CODE REVIEW

File: src/reddit.ts [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+      sr: z.string(), // The subreddit to submit to. Ex: "api_reference"
+      kind: z
+        .enum(["link", "self", "image", "video", "videogif"])
+        .default("self"), // Specify the type of post. https://www.reddit.com/dev/api#POST_api_submit
+    }),
+  }),
+  run: async (payload, io, ctx) => {
+    // Get the access token
+    // First time we need to get the access token
+    if (!requestOptions) {
```
Comment (nicktrn) [inline src/reddit.ts:48]: This won't work as expected. Please don't try to set global vars from a job run.

I think we will have to get the access token each time until we have an integration for this. Also best to stick this in a separate task.

You would also have to deal with [refreshing](https://github.com/reddit-archive/reddit/wiki/OAuth2#refreshing-the-token) or see what happens if you just create a new token each time the job runs - there may be limits etc.

**Comment by biplobsd** [comment]
Hello @nicktrn kindly review again. I have used subtasks. Is it right?

**Comment by nicktrn** [comment]
Thanks for those edits! I think it should be fine to remove the outer `io.runTask()`. Unless you think it's needed?

More importantly though, what happens with the `access_token` if we have multiple jobs running in parallel? Might be fine, but would have to be tested.

Might run into issues here that are unable to be resolved without a proper integration, no worries if that's the case. We may just have to leave the example for now.

**Comment by biplobsd** [comment]
> Thanks for those edits! I think it should be fine to remove the outer `io.runTask()`. Unless you think it's needed?
> 
> More importantly though, what happens with the `access_token` if we have multiple jobs running in parallel? Might be fine, but would have to be tested.
> 
> Might run into issues here that are unable to be resolved without a proper integration, no worries if that's the case. We may just have to leave the example for now.

I saw this approach in the task [docs.](https://trigger.dev/docs/documentation/concepts/tasks#subtasks) I don't know it's best or not.. but in this example maybe user learn more about subtask?

**Comment by biplobsd** [comment]
Thanks @D-K-P 🎉✨

**Pull request merged**